### ISSUE TITLE
Auto-Intsrumentation not working!

### ISSUE THREAD

**Issue by hnkisadiq** [signalfx/splunk-otel-python#98]
I have a simple flask app for health check, I am trying to auto instrument it but I get an error saying:
 **Traces cannot be uploaded; HTTP status code: 500, message: Internal Server Error**

I have configured the env variables appropriately and I am using jeager-thrift-splunk exporter.
**export OTEL_SERVICE_NAME=sfxscripts 
export OTEL_TRACES_EXPORTER=jaeger-thrift-splunk
export OTEL_EXPORTER_JAEGER_ENDPOINT=http://<<IP>>:14268/api/traces**

Also, I notice that some examples provided are outdated and are throwing version incompatibility error. -- https://github.com/signalfx/tracing-examples/tree/main/opentelemetry-tracing/opentelemetry-python-tracing/flask-and-uswgi

I am running it like this: `splunk-py-trace python app.py`

**Code:**
```
import requests

from flask import Flask
app = Flask(__name__)

@app.route('/health')
def health():
    isHealthy = False
    status_healthy = "OK"
    status_unhealthy = "unhealthy"

    try:
        response = requests.get("http://localhost:13133/")
        if response.status_code != requests.codes.ok:
            print("Otelcollector might not be running - " % response)
        else:
            responseInJson = response.json()
            if "status" in responseInJson and responseInJson["status"] == "Server available":
                isHealthy = True

    except Exception as e_message:
        print("Encountered error while checking Otelcollector health = %s" % e_message)

    return status_healthy if isHealthy else status_unhealthy

if __name__ == '__main__':
    app.run(debug=False, host = '0.0.0.0', port=8083)
```

**Comment by owais** [comment]
Thanks for reporting the issue. I tried to replicate it but could not. I tried with the latest version `0.15.2` and exporter to Jaeger "all in one" (https://hub.docker.com/r/jaegertracing/all-in-one). All traces were exporter properly to the locally running Jaeger instance. 

What exactly are you trying to export the traces to? It looks like the service you are sending traces to is not accepting them and returning HTTP 500 errors.

**Comment by hnkisadiq** [comment]
Hey! Thanks for replying..
I am trying to export it to Signalfx (Splunk Observability) via an otel endpoint. 
when I try manually instrumenting with opentelemetry python through jaeger exporter it works fine and I can see the traces on signalfx but auto instrumentation is not happening.

**Comment by owais** [comment]
SignalFx ingest supports Jaeger, Zipkin and SAPM today. It does not support OTLP gRPC. It support OTLP HTTP but OpenTelemetry Python does not have that exporter available yet. You should be able to export to SignalFx ingest if you use `jaeger-thrift-splunk` exporter and set the endpoint to your org's ingest endpoint. You'll also need to set the `SPLUNK_ACCESS_TOKEN` env var.

**Comment by sparam** [comment]
Hi @owais ,

@hnkisadiq  and I were trying to debug regarding the error `Traces cannot be uploaded; HTTP status code: 500, message: Internal Server Error`. Below environment vars are set. 

export OTEL_SERVICE_NAME=sfxscripts
export OTEL_TRACES_EXPORTER=jaeger-thrift-splunk
export OTEL_EXPORTER_JAEGER_ENDPOINT=http://<>:14268/api/traces

**Comment by hnkisadiq** [comment]
@owais I tried setting SPLUNK_ACCESS_TOKEN var as well, but no luck..

**Comment by hnkisadiq** [comment]
`import requests
from splunk_otel.tracing import start_tracing
from opentelemetry.instrumentation.flask import FlaskInstrumentor
from flask import Flask
app = Flask(__name__)

start_tracing()
# instrument our flask app instance eplicitly
FlaskInstrumentor().instrument_app(app)

@app.route('/health')
def health():
    isHealthy = False
    status_healthy = "OK"
    status_unhealthy = "unhealthy"

    try:
        response = requests.get("http://localhost:13133/")
        if response.status_code != requests.codes.ok:
            print("Otelcollector might not be running - " % response)
        else:
            responseInJson = response.json()
            if "status" in responseInJson and responseInJson["status"] == "Server available":
                isHealthy = True

    except Exception as e_message:
        print("Encountered error while checking Otelcollector health = %s" % e_message)

    return status_healthy if isHealthy else status_unhealthy

if __name__ == '__main__':
    app.run(debug=False, host = '0.0.0.0', port=8084)`

**Comment by hnkisadiq** [comment]
@owais I even tried the above way but still not able to see any traces

**Comment by hnkisadiq** [comment]
is it because that its a flask app that I need opentelemetry.instrumentation.flask to instrument it?

**Comment by owais** [comment]
Did you manually install the `opentelemetry-instrumentation-flask` package? You're right that `splunk-opentelemetry[all]` does not include flask (or any other) instrumentation. Instrumentation packages need to be installed after by running the `splunk-py-trace-bootstrap` command.

If you remove the below lines but keep the instrumentation package installed, run the program with `splunk-py-trace python app.py` and see if it works?

```
from splunk_otel.tracing import start_tracing
from opentelemetry.instrumentation.flask import FlaskInstrumentor
```

```
start_tracing()
# instrument our flask app instance eplicitly
FlaskInstrumentor().instrument_app(app)
```

**Comment by hnkisadiq** [comment]
@owais Yes it works... perhaps I wasnt clear in my previous message but thats what I meant when I said that auto instrumentation is working.. I removed those lines and tried with `splunk-py-trace python app.py`.

Also I did run `splunk-py-trace-bootstrap` but flask-instrumentation was not installed..

Thanks much for you help! Also it would be nice if this was explicitly mentioned in the readme file.
Closing this issue now.
